### PR TITLE
chore: Setting up running on python 3.10

### PR DESCRIPTION
**Pull request by devinmatte**
*Ensure your PR title follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) spec*

## Motivation

Ensure our whole stack of apps runs on the same version of python. Allow 3.10 testing for a phase 2 full upgrade to python 3.10

## Changes

- Allows python 3.10 in deps and CI
- Updates dependencies for upgrade

## Testing Instructions

Dry run is successful in 3.10

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -9,15 +9,19 @@ on:
...
         uses: actions/setup-python@v3
         with:
-          python-version: 3.9
+          python-version: ${{ matrix.python-version }}
       - name: Lint backend code with Flake8
         run: |
           curl -sSL https://install.python-poetry.org | python3 -
           poetry install
           poetry run flake8
+          poetry run black --check .
```
Comment (nathan-weinberg) [inline .github/workflows/lint.yml:27]: Why are we running both `flake8` and `black`?

**Review by nathan-weinberg — COMMENTED** [review]
One question but otherwise LGTM

File: .github/workflows/lint.yml (hunk 1 above)
Comment (devinmatte) [reply .github/workflows/lint.yml:27]: Flake8 is linting, black is formatting

Comment (nathan-weinberg) [reply .github/workflows/lint.yml:27]: Gotcha, maybe then we can change the name of this from `Lint backend code with Flake8` to `Lint and check formatting of backend code` or something?

**Review by nathan-weinberg — APPROVED** [review]

**Pull request merged**